### ISSUE TITLE
Add group members to restricted groups

### ISSUE THREAD

**Issue by ajpeddakotla** [hypothesis/product-backlog#494]
### User story: 
As a Hypothesis admin (H team member), I want to be able create a restricted group (covered in the admin interface card: #421), and add/edit/remove group members from the restricted group. 

### Acceptance Criteria: 
- [ ] group members can be added by entering a list of H usernames (for already existing users) 
- [ ] group members can be removed from the group (they will not receive a notification that they have been removed from the group) 
- [ ] H admins can see a list of current group members within a group and the number of annotations that the member has made.
- [ ] After creating a new group using the admin page the Admin should be taken to a panel to manage the membership of the group

**NOTE:** 
- No design elements are required as part of this work since we will be using the existing forms present on the admin site. 
- @gobengo's work is very applicable here, and can serve as a starting point for some ideation on this card.

**Comment by lyzadanger** [comment]
Note that this card is dependent on #421

**Comment by seanh** [comment]
> * group members can be added by entering a list of emails
>  * entering an email will send an email to that person, where they can click the link the email to create a Hypothesis account and join the group.

Really? Are we implementing a new email-based invite to create a Hypothesis account and join a group flow here? That will be a lot of work. Maybe as a first pass the Hypothesis admin should just be able to add existing Hypothesis users to the group? And it's up to users to register and activate their own accounts beforehand.

Invitations are a great feature but definitely a big, separate thing on its own

**Comment by seanh** [comment]
> group members information can be edited: username, and the email associated with their account.

This seems like a separate feature to me (giving Hypothesis admins the ability to edit the usernames and email addresses of users).

I'd suggest keeping it simple at first: just an admin page for adding existing Hypothesis users to groups, and removing them from groups.

**Comment by ajpeddakotla** [comment]
Follow ups: 

> group members can be added by entering a list of emails
entering an email will send an email to that person, where they can click the link the email to create a Hypothesis account and join the group.

> Really? Are we implementing a new email-based invite to create a Hypothesis account and join a group flow here? That will be a lot of work. Maybe as a first pass the Hypothesis admin should just be able to add existing Hypothesis users to the group? And it's up to users to register and activate their own accounts beforehand.
Invitations are a great feature but definitely a big, separate thing on its own

After going through the existing login/sign up flow for users that are sent share/join links to current groups, I realized that this won't work for restricted groups b/c we're changing how that link works. 
So yes @seanh group members can only be added (for now) by entering H usernames. 

But question - what will happen after I enter in a username? How will the user be notified that they have been added to the group? 

_________
>group members information can be edited: username, and the email associated with their account.
This seems like a separate feature to me (giving Hypothesis admins the ability to edit the usernames and email addresses of users).
I'd suggest keeping it simple at first: just an admin page for adding existing Hypothesis users to groups, and removing them from groups.

I do think this is a useful thing to add for publishers who won't (right away) have admin control over their groups (they won't have an admin dashboard they can go to for user management). So this really in anticipation of the requests that we might get - but I do agree that it is a seperate ticket, so I'll take it out of this issue.

**Comment by seanh** [comment]
> But question - what will happen after I enter in a username? How will the user be notified that they 
> have been added to the group?

My suggestion at first is that, since Hypothesis staff admins are gonna be the only people adding users to restricted groups, the admin can just email the user manually if necessary?

I think we should have a separate card for automated email notifications. Hypothesis could automatically send out a "You've been added to a group" email with a link to the group's page, for example. But I think we should capture this on a separate card and prioritize it appropriately. Possible user stories:

* As a user I want to be notified when I've been added to a restricted group so that I know when I can start annotating in the group
* As a Hypothesis staff admin I want users to be notified automatically when I add them to a restricted group so that I don't have to spend time notifying them manually

**Comment by seanh** [comment]
> After going through the existing login/sign up flow for users that are sent share/join links to current 
> groups, I realized that this won't work for restricted groups b/c we're changing how that link works.

Yes. I've been trying to raise this issue with restricted groups. (It's explained on the description of the  [restricted groups epic card](https://github.com/hypothesis/product-backlog/issues/424), for example.) With private groups we use the group's home page also as the invite link, but for restricted groups this won't work since the group's home page is public. If we want to implement invites for restricted groups one day then we're going to have to design a better invite system.

Private groups could also probably benefit from a better invite flow themselves too.

**Comment by seanh** [comment]
> > > group members information can be edited: username, and the email associated with their account.
> > >
> > This seems like a separate feature to me (giving Hypothesis admins the ability to edit the usernames
> > and email addresses of users).
> > I'd suggest keeping it simple at first: just an admin page for adding existing Hypothesis users to 
> > groups, and removing them from groups.
> > 
> I do think this is a useful thing to add for publishers who won't (right away) have admin control over 
> their groups (they won't have an admin dashboard they can go to for user management). So this really 
> in anticipation of the requests that we might get - but I do agree that it is a seperate ticket, so I'll take it 
> out of this issue.

Agree, it does seem like a useful thing to have but can go on a separate card (and then be prioritized separately too)
